src/output.rs

```diff
@@ -21,11 +38,92 @@ pub fn print_activation(activated: bool) {
     eprintln!("\x1b[1;34m{} {}.\x1b[0m", word, SHADOWENV);
 }
 
-fn backticks_to_bright_green(err: failure::Error) -> String {
+fn backticks_to_bright_green(err: Error) -> String {
     let re = Regex::new(r"`(.*?)`").unwrap();
     // this is almost certainly not the best way to do this, but this runs at most once per
     // execution so I only care so much.
     let before = format!("{}", err);
     re.replace_all(before.as_ref(), "\x1b[1;32m$1\x1b[1;31m")
         .to_string()
 }
+
+fn check_and_trigger_cooldown(err: &Error) -> Result<bool, Error> {
+    // if no .shadowenv.d, then Err(_) just means no cooldown: always display error.
+    let root = loader::find_root(env::current_dir()?, loader::DEFAULT_RELATIVE_COMPONENT)?
+        .ok_or_else(|| format_err!("no .shadowenv.d"))?;
+
+    let _ = clean_up_stale_errors(&root, Duration::new(300, 0));
```

Comment:
Mainly, I wanted the benefit of writing that function with `?`, but there's no real situation where I'd want to interrupt a shell hook because it failed to clean up some files.

There has to be a better pattern here though.